horovod/tensorflow/__init__.py

```diff
@@ -202,7 +219,18 @@ def after_create_session(self, session, coord):
 
 @_cache
 def _make_allreduce_grads_fn(name, device_dense, device_sparse,
-                             compression, sparse_as_dense, op):
+                             compression, sparse_as_dense, op, gradient_predivide_factor):
+    if op == Average and gradient_predivide_factor != 1.0:
+        # Perform averaging via pre/postscaling factors.
+        # Split average operation across pre/postscale factors
+        prescale_factor = 1.0 / gradient_predivide_factor
+        postscale_factor = gradient_predivide_factor / size()
+        true_op = Sum
+    else:
```

Comment:
The user isn't in direct control of the prescale and postscale factors in this part of the code, since it is within the `DistributedOptimizer`. Their only mechanism to change these here is through `gradient_predivide_factor`. I did add a new exception to ensure that `gradient_predivide_factor` is only used if `op==Average`. 

If a user wants to apply pre- and post-scale factors with other operations (sum, adasum), they can do so using the normal `allreduce` function.